client/my-sites/purchases/controller.js

```diff
@@ -23,3 +23,10 @@ export function redirectToPurchases( context ) {
 
 	return page.redirect( '/purchases' );
 }
+
+export const purchaseSettings = ( context, next ) => {
+	context.primary = (
+		<PurchaseSettings siteId={ context.params.site } receiptId={ context.params.receiptId } />
```

Comment:
I know this is still in-progress, but you might find using the `getSelectedSiteId` Redux selector more effective than passing it from context here because it will be normalized to an ID that way; here, `context.params.site` might be a number or a `example.wordpress.com` string or a `example.com` domain.